django/db/models/expressions.py

```diff
@@ -1859,23 +1867,37 @@ class WindowFrame(Expression):
 
     template = "%(frame_type)s BETWEEN %(start)s AND %(end)s"
 
-    def __init__(self, start=None, end=None):
+    def __init__(self, start=None, end=None, exclusion=None):
         self.start = Value(start)
         self.end = Value(end)
+        self.exclusion = exclusion
 
     def set_source_expressions(self, exprs):
         self.start, self.end = exprs
 
     def get_source_expressions(self):
         return [self.start, self.end]
 
+    def get_exclusion(self):
+        if not connection.features.supports_frame_exclusion:
+            raise NotSupportedError(
+                "This backend does not support window frame exclusions."
+            )
+        if self.exclusion in Exclusion.__members__.values():
+            return f"EXCLUDE {self.exclusion.value}"
+        raise ValueError("exclusion must be a valid Exclusion value.")
+
     def as_sql(self, compiler, connection):
         connection.ops.check_expression_support(self)
         start, end = self.window_frame_start_end(
             connection, self.start.value, self.end.value
         )
+        window_frame_template_parts = [self.template]
+        if self.exclusion is not None:
+            exclusion_as_sql = self.get_exclusion()
+            window_frame_template_parts.append(exclusion_as_sql)
         return (
-            self.template
+            " ".join(window_frame_template_parts)
```

Comment:
What do you think about adding `exclude` to the `self.template` instead?
`template = "%(frame_type)s BETWEEN %(start)s AND %(end)s%(exclude)s"`
```
def get_exclusion(self):
    ...
    if isinstance(self.exclusion, Exclusion):
        return f" EXCLUDE {self.exclusion.value}"
```